In [25]:
#Confidence Intervals Using the t Distribution
#The following example shows how to calculate a confidence interval for the true population mean height (in inches) of a certain species of plant, using a sample of 15 plants:
#If we’re working with a small sample (n <30), we can use the t.interval() function from the scipy.stats library to calculate a confidence interval for a population mean.

import numpy as np
import scipy.stats as st

#define sample data
data = [12, 12, 13, 13, 15, 16, 17, 22, 23, 25, 26, 27, 28, 28, 29]

#create 95% confidence interval for population mean weight. You’ll notice that the larger the confidence level, the wider the confidence interval.
print(st.t.interval(confidence = 0.95, df=len(data)-1, loc=np.mean(data), scale=st.sem(data)))
print(st.t.interval(confidence = 0.80, df=len(data)-1, loc=np.mean(data), scale=st.sem(data)))


(np.float64(16.75776979778349), np.float64(24.04223020221651))
(np.float64(18.115898611113977), np.float64(22.68410138888602))


That is, there’s only a 5% chance that the true population mean height of plants is less than 16.758 inches or greater than 24.042 inches.

In [26]:
#Confidence Intervals Using the Normal Distribution
#If we’re working with larger samples (n≥30), we can assume that the sampling distribution of the sample mean is normally distributed
#(thanks to the Central Limit Theorem) and can instead use the norm.interval() function from the scipy.stats library.


#define sample data
np.random.seed(0)
data = np.random.randint(10, 30, 50)

#create 95% confidence interval for population mean weight
print(st.norm.interval(confidence = 0.95, loc=np.mean(data), scale=st.sem(data)))
print(st.norm.interval(confidence = 0.8, loc=np.mean(data), scale=st.sem(data)))


(np.float64(17.400060940568054), np.float64(21.079939059431943))
(np.float64(18.03692851464563), np.float64(20.443071485354366))


In [27]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

df = pd.read_excel(open("Descriptive-statistics-HomePrice.xlsx", "rb"), "Data")
df.head()

/Users/qiyunge/winMac/SoftwareDev/LaselleCollege/data_science/.venv3.12/lib/python3.12/site-packages/openpyxl/worksheet/_reader.py:329: UserWarning: Data Validation extension is not supported and will be removed
  warn(msg)


,ID,Building,Year of sale,Month of sale,Type of property,Property #,Area (ft.),Price,Status,Customer ID,...,Y,M,D,Gender,Country,State,Purpose,Deal satisfaction,Mortgage,Source
0,1030,1,2005.0,11.0,Apartment,30,743.0856,246172.6760,Sold,C0028,...,1986.0,6.0,21.0,F,USA,California,Home,5.0,No,Website
1,1029,1,2005.0,10.0,Apartment,29,756.2128,246331.9040,Sold,C0027,...,1983.0,2.0,24.0,F,USA,California,Home,5.0,No,Website
2,2002,2,2007.0,7.0,Apartment,2,587.2808,209280.9104,Sold,C0112,...,1985.0,12.0,27.0,M,USA,California,Home,1.0,Yes,Client
3,2031,2,2007.0,12.0,Apartment,31,1604.7464,452667.0064,Sold,C0160,...,1985.0,12.0,27.0,M,USA,California,Investment,3.0,Yes,Website
4,1049,1,2004.0,11.0,Apartment,49,1375.4508,467083.3132,Sold,C0014,...,1979.0,5.0,15.0,F,USA,California,Home,4.0,No,Agency


In [28]:
#99% confidence interval for price mean
data = df['Price']
st.t.interval(confidence = 0.99, df=len(data)-1, loc=np.mean(data), scale=st.sem(data))

(np.float64(267021.8320262084), np.float64(295321.97097603884))

In [29]:
#Create a mple from our population
sampled_df = df.sample(n=200, random_state=100)
#optional: I used the whole dataset
# sampled_df = df

In [30]:
# Remove Empty Value from List
sampled_df.dropna(inplace=True)

In [31]:
WebsiteBuyer = sampled_df[sampled_df['Source']=='Website']['Age at time of purchase']
ClientBuyer = sampled_df[sampled_df['Source']=='Client']['Age at time of purchase']
print(WebsiteBuyer.shape)
print(ClientBuyer.shape)


(70,)
(12,)


In [32]:
WebsiteBuyer

83     48
111    50
97     47
46     37
115    51
       ..
36     34
43     37
131    55
0      19
78     49
Name: Age at time of purchase, Length: 70, dtype: object

In [33]:
femaleBuyer = sampled_df.loc[(sampled_df['Source']=='Website') & (sampled_df['Gender']=='F')]['Age at time of purchase']
print(femaleBuyer.shape)
maleBuyer = sampled_df.loc[(sampled_df['Source']=='Website') & (sampled_df['Gender']=='M')]['Age at time of purchase']
print(maleBuyer.shape)
femaleBuyer = pd.to_numeric(
    sampled_df.loc[
        (sampled_df['Source'] == 'Website') & (sampled_df['Gender'] == 'F'),
        'Age at time of purchase'
    ],
    errors='coerce'
).dropna()

maleBuyer = pd.to_numeric(
    sampled_df.loc[
        (sampled_df['Source'] == 'Website') & (sampled_df['Gender'] == 'M'),
        'Age at time of purchase'
    ],
    errors='coerce'
).dropna()

(28,)
(42,)


In [34]:
#from scipy.stats import t
rv = st.t(df=maleBuyer.shape[0] + femaleBuyer.shape[0] -2)

In [35]:
#from scipy.stats import ttest_ind
t_stat,pvalue = st.ttest_ind(femaleBuyer,maleBuyer)
print(pvalue,t_stat)

0.5718549947777972 -0.5680752308187156


In [36]:
FemaleBuyer = sampled_df[sampled_df['Gender']=='F']['Price']
MaleBuyer = sampled_df[sampled_df['Gender']=='M']['Price']
print(FemaleBuyer.shape)
print(MaleBuyer.shape)

(46,)
(71,)


In [37]:
rv = st.t(df= FemaleBuyer.shape[0] + MaleBuyer.shape[0]-2)
t_stat, pvalue = st.ttest_ind(FemaleBuyer, MaleBuyer)
pvalue

np.float64(0.3006043250932705)

In [41]:
webAge = sampled_df.loc[(sampled_df['Source']=='Website')]['Age at time of purchase'].dropna().astype("Int32")
print(webAge.shape)
agencyAge = sampled_df.loc[(sampled_df['Source']=='Client')]['Age at time of purchase'].dropna().astype("Int32")
print(agencyAge.shape)
rv = st.t(df=webAge.shape[0] + agencyAge.shape[0]-2)
t_stat, pvalue = st.ttest_ind(webAge, agencyAge)
pvalue

(70,)
(12,)


np.float64(0.22466639215628403)

In [ ]:
# area ? price